.github/workflows/tests.yml

```diff
@@ -37,7 +40,34 @@ jobs:
             const { issue: { number: issue_number }, repo: { owner, repo } } = context;
             const label = 'run-tests';
             github.issues.removeLabel({ owner, repo, issue_number, name: label });
-
+      - name: Checkout Amplify UI
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ steps.get-ref.outputs.ref }}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          persist-credentials: false
+      - name: Setup Node.js LTS
+        uses: actions/setup-node@v2
+        with:
+          node-version: lts/*
+          cache: 'yarn'
+      - name: Get ref and commit
+        id: get-ref
+        run: |
+          echo "::set-output name=key::${{ github.event_name == 'push' && github.ref || github.event.pull_request.head.ref }}"
+          echo "::set-output name=commit::$(git rev-parse --short "$GITHUB_SHA")"
+      - name: Install packages
+        run: yarn --frozen-lockfile
```

Comment:
We should look into separating `build` from `install`, at least from CI side. This will make more rooms for optimization.

Reply:
Feel free to actually remove it here – my previous attempts separated them, too: #396 

We'll just need to update the `CONTRIBUTING.md` and maybe the `.husky` hook to build after `git pull`.